backend/module/chaos/serverexperimentation/rtds/rtds_cache_test.go

```diff
@@ -209,57 +237,59 @@ func TestCreateRuntimeKeys(t *testing.T) {
 
 	for _, testExperiment := range testDataList {
 		var expectedPercentageKey string
-		var expectedPercentageValue float32
+		var expectedPercentageValue uint32
 		var expectedFaultKey string
-		var expectedFaultValue int32
+		var expectedFaultValue uint32
 
-		config := &serverexperimentation.TestConfig{}
+		config := &serverexperimentation.HTTPFaultConfig{}
 		err := ptypes.UnmarshalAny(testExperiment.GetConfig(), config)
 		if err != nil {
 			t.Errorf("unmarshalAny failed %v", err)
 		}
 
-		target := config.GetClusterPair()
-		faultInjectionType := target.GetFaultInjectionCluster()
+		upstream, downstream, err := unParse(config)
+		assert.NoError(t, err)
 
 		switch config.GetFault().(type) {
-		case *serverexperimentation.TestConfig_Abort:
-			abort := config.GetAbort()
-			expectedFaultValue = abort.HttpStatus
-			expectedPercentageValue = abort.Percent
-
-			if faultInjectionType == serverexperimentation.FaultInjectionCluster_FAULTINJECTIONCLUSTER_DOWNSTREAM {
-				expectedPercentageKey = fmt.Sprintf(HTTPPercentageForExternal, egressPrefix, target.UpstreamCluster)
-				expectedFaultKey = fmt.Sprintf(HTTPStatusForExternal, egressPrefix, target.UpstreamCluster)
-			} else if faultInjectionType == serverexperimentation.FaultInjectionCluster_FAULTINJECTIONCLUSTER_UPSTREAM {
-				if target.DownstreamCluster == "" {
+		case *serverexperimentation.HTTPFaultConfig_AbortFault:
+			abort := config.GetAbortFault()
+			expectedFaultValue = abort.GetAbortStatus().GetHttpStatusCode()
+			expectedPercentageValue = abort.GetPercentage().GetPercentage()
+
+			switch interface{}(config.GetFaultTargeting().GetEnforcer()).(type) {
```

Comment:
It might be nice to add some extra helper functions here and elsewhere just to reduce some of the nesting in this function, it's a bit hard to follow

Reply:
This nesting has different logics within it. I could make a func for this specific case which can be used in `rtds_cahce.go` but I don't see much benefit from it. Let me know your thought